pkg/ast/pipesearch/searchHandler.go

```diff
@@ -268,6 +268,10 @@ func ProcessAlertsPipeSearchRequest(queryParams alertutils.QueryParams) int {
 			// Dedup needs to see all the matched records before it can return any
 			// of them when there's a sortby option.
 			sizeLimit = math.MaxUint64
+		} else if aggs.HasRexBlockInChainWithStats() {
+			// If there's a Stats block in the chain followed by a Rex block, we need to
+			// see all the matched records before we apply or calculate the stats.
+			sizeLimit = math.MaxUint64
```

Comment:
nit: I think this condition is implemented correctly, but the comment is slightly off. You're checking if there's a rex block followed by stats, not a stats block followed by rex